src/Core/src/Hosting/MauiServiceProvider.cs

```diff
@@ -66,7 +69,12 @@ public MauiServiceProvider(IMauiServiceCollection collection)
 		object? CreateInstance(ServiceDescriptor item)
 		{
 			if (item.ImplementationType != null)
-				return Activator.CreateInstance(item.ImplementationType);
+			{
+				if (_constructorInjection)
+					return CreateInstance(item.ImplementationType);
+				else
+					return Activator.CreateInstance(item.ImplementationType);
```

Comment:
There is always the raw Activator if need be or if constructor injection was not requested.